internal/service/iam/instance_profiles_data_source_test.go

```diff
@@ -0,0 +1,58 @@
+package iam_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/iam"
+	sdkacctest "github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+)
+
+func TestAccIAMInstanceProfilesDataSource_basic(t *testing.T) {
+	datasourceName := "data.aws_iam_instance_profiles.test"
+	resourceName := "aws_iam_instance_profile.test"
+
+	roleName := fmt.Sprintf("tf-acc-ds-instance-profile-role-%d", sdkacctest.RandInt())
+	profileName := fmt.Sprintf("tf-acc-ds-instance-profile-%d", sdkacctest.RandInt())
```

Comment:
We may use `sdkacctest.RandomWithPrefix` instead as it seems to be a better practice.
https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/running-and-writing-acceptance-tests.md#randomized-naming